cmd/po-docgen/operator.go

```
@@ -0,0 +1,226 @@
+		return exprIdent.String()
+	}
+
+	retValue := ""
+
+	if exprIdent.Obj.Kind == ast.Con {
+		valSpec := exprIdent.Obj.Decl.(*ast.ValueSpec)
+		for _, value := range valSpec.Values {
+			retValue += unquoteLiteral(value.(*ast.BasicLit).Value)
+		}
+	}
+	return retValue
+
+}
+
+func resolveBoolExpr(expr ast.Expr) string {
+	exprIdent, ok := expr.(*ast.Ident)
+	if ok {
+		return exprIdent.Name
+	}
+
+	return "No Identifier to Resolve"
+}
+
+func resolveConstStringExpr(expr ast.Expr) string {
+
+	exprBasicLit, ok := expr.(*ast.BasicLit)
+	if ok {
+		return unquoteLiteral(exprBasicLit.Value)
+	}
+
+	exprIdent, ok := expr.(*ast.Ident)
+	if ok {
+		retValue := ""
+		if exprIdent.Obj.Kind == ast.Con {
+			valSpec := exprIdent.Obj.Decl.(*ast.ValueSpec)
+			for _, value := range valSpec.Values {
+				retValue += unquoteLiteral(value.(*ast.BasicLit).Value)
+			}
+		}
+		return retValue
+	}
+
+	exprBinaryOp, ok := expr.(*ast.BinaryExpr)
+	if ok {
+		operandX := resolveConstStringExpr(exprBinaryOp.X)
+		operandY := resolveConstStringExpr(exprBinaryOp.Y)
+		if exprBinaryOp.Op == token.ADD {
+			return operandX + operandY
+		}
+		return fmt.Sprint("unhandled OP", exprBinaryOp.Op)
+	}
+
+	return ""
+}
+
+func operatorCodeToDoc(paths []string) []FlagDoc {
+	astPkg := fileToASTPackage(paths)
+	flagDocs := make([]FlagDoc, 0)
+
+	for _, file := range astPkg.Files {
+		fmt.Println(file.Name.Name)
+		for _, decl := range file.Decls {
+			funcdecl, ok := decl.(*ast.FuncDecl)
+			if ok {
+				if funcdecl.Name.Name == "init" {
+					for _, statement := range funcdecl.Body.List {
+						exprStatement, ok := statement.(*ast.ExprStmt)
+						if !ok {
+							continue
+						}
+						exprCall, ok := exprStatement.X.(*ast.CallExpr)
+						if !ok {
+							continue
+						}
+
+						if exprCall.Fun.(*ast.SelectorExpr).X.(*ast.Ident).Name == "flagset" {
+							var argName, argDefaultValue, argDescription string
+							if exprCall.Fun.(*ast.SelectorExpr).Sel.Name == "StringVar" {
```

Shouldn't it fail hard if it encounters a flag type that isn't supported by the generator? Also a switch/case would be more [idiomatic](https://tour.golang.org/flowcontrol/9).

```
selectorExpr, ok := exprCall.Fun.(*ast.SelectorExpr)
if !ok {
    continue
}

switch selectorExpr.Sel.Name {
    case "StringVar":
    ...
    default:
        // return an error
}
```